packages/engine/Source/Core/Resource.js

```diff
@@ -319,13 +142,30 @@ function Resource(options) {
   this.retryAttempts = defaultValue(options.retryAttempts, 0);
   this._retryCount = 0;
 
-  const uri = new Uri(options.url);
-  parseQuery(uri, this, true, true);
+  const parseUrl = defaultValue(options.parseUrl, true);
+  if (parseUrl) {
+    this.parseUrl(options.url, true, true);
+  } else {
+    this._url = options.url;
+  }
+}
 
-  // Remove the fragment as it's not sent with a request
-  uri.fragment("");
+/**
+ * Clones a value if it is defined, otherwise returns the default value
+ *
+ * @param {*} [val] The value to clone.
```

Comment:
`{Resource}` for all of these?

Also, we try not to use abbreviations (ie use `value` instead of `val`).

Reply:
This helper is mostly used to process options objects, without defined types. I cleaned up the naming, and also deleted the check for `val.clone`, since none of the values passed to the function had such a method.